src/browser/engine/protocol.ts

```diff
@@ -1,58 +1,81 @@
 import { app, protocol } from 'electron';
-import { readFile } from 'fs';
+import { readFileSync, createReadStream } from 'fs';
+import stream from 'stream';
 import { lookup } from 'mime-types';
 import { join } from 'path';
 import { parse } from 'url';
+import ECx from './api';
 
 import { Protocol } from '../../common';
 import { protocolAsScheme } from '../../common/utils';
 import { getExtensionById } from '../chrome-extension';
 
+// tslint:disable-next-line: max-line-length
+const defaultContentSecurityPolicy = 'script-src \'self\' blob: filesystem: chrome-extension-resource:; object-src \'self\' blob: filesystem:;';
+
 protocol.registerStandardSchemes(
   [protocolAsScheme(Protocol.Extension)],
   { secure: true }
 );
 
-const protocolHandler = (
+const protocolHandler = async (
   { url }: Electron.RegisterBufferProtocolRequest,
   callback: Function
 ) => {
   const { hostname, pathname } = parse(url);
   if (!hostname || !pathname) return callback();
 
   const extension = getExtensionById(hostname);
-
   if (!extension) return callback();
 
   const { src, backgroundPage: { name, html } } = extension;
+  const headers = {};
+
+  // Hack to get extension ID
+  const splitted = src.split('/');
+  const ecxId = splitted[splitted.length - 2];
```

Comment:
This is a hack, while the hostname is temporarily modified to a human readable name and doesn't contain the chrome extension ID anymore.

Reply:
no prob ;)

Reply:
mark as todo